bite-size-notes/week-1-a-discretization-numerical-integration-methods.md

```diff
@@ -0,0 +1,148 @@
+
+A Taylor series expansion is a representation of a function as an infinite sum of terms, where each term is a polynomial evaluated at a specific point. The Taylor series expansion provides an approximation of the original function near that point.
+
+The general form of a Taylor series expansion for a function $f(x)$ around the point $a$ is given by:
+
+$$
+f(x) = f(a) + f'(a)(x-a) + \frac{f''(a)}{2!}(x-a)^2 + \frac{f'''(a)}{3!}(x-a)^3 + \cdots 
+$$
+
+where:
+
+- $f(a)$ is the value of the function at the point $a$,
+- $f'(a)$ is the first derivative of the function evaluated at $a$,
+- $f''(a)$ is the second derivative of the function evaluated at $a$,
+- $f'''(a)$ is the third derivative of the function evaluated at $a$, and so on,
+- $(x-a)$ represents the deviation from the point $a$,
+- $n!$ denotes the factorial of $n$.
+
+The terms beyond the linear term $f'(a)(x-a)$ involve higher-order derivatives of the function and higher powers of $(x-a)$, capturing increasingly finer details of the function's behavior near the point $a$.
+
+By truncating the Taylor series expansion at a certain order, one can obtain a polynomial approximation of the function around the point $a$. This idea can be viewed in the Euler's method (truncating at the first-order) and Runge-Kutta method (truncating at the fourth-order). The accuracy of the approximation depends on the smoothness of the function and the distance from the point $a$.
+
+*Reference: https://en.wikipedia.org/wiki/Taylor_series*
+
+## Euler's Method
+
+![Euler's Method Image](figs/euler_method.png "https://m0nads.wordpress.com/tag/euler-method/")
+
+*image: https://m0nads.wordpress.com/tag/euler-method/*
+
+The Euler method is a first-order method for solving ODEs with a given value. It is the most basic explicit method for numerical intergration of ODEs and is simplest Runge-Kutta method
+
+**Mathematical Formulation:**
+
+$$x_{k+1} = x_{k} + \dot{x}(t(k))\Delta t$$
+
+where,
+
+$$\Delta t \text{ (or } h \text{) is a constant step size}$$
+
+### Advantages and Limitations:
+
+- **Advantages**
+    - Simple and easy to implement
+    - Requires only basic mathematical operations
+    suiteable for problems where analytical solutions are unavasilable or impractical.
+
+- **Disadvantages**
+    - May not provide accurate results for highly nonlinear or stiff ODEs.
+    - Approximation eeor is proportional to the step size, $\Delta t$, chosen. Good approximation is obtained with very small $\Delta t$ but would require a large number of time discretixations leading to a large computation time.
+
+### When is it used?:
+
+Euler's method can be used when the system does not grow too quickly.
+
+Euler's method is rarely used in real-world applications as the algorithm tends to have low accuracy and requires vast computation time.
+
```

Comment:
Showing advantages and disadvantages is helpful to understand! I like this structure.